diff --git a/Library/Homebrew/dev-cmd/bump.rb b/Library/Homebrew/dev-cmd/bump.rb
index 0bc0bbc96..e00b2a68a 100644
--- a/Library/Homebrew/dev-cmd/bump.rb
+++ b/Library/Homebrew/dev-cmd/bump.rb
@@ -281,7 +281,7 @@ module Homebrew
     return unless new_version
     return if pull_requests
 
-    system HOMEBREW_BREW_FILE, "bump-#{type}-pr", "--no-browse",
+    system HOMEBREW_BREW_FILE, "bump-#{type}-pr", "--no-browse", "--no-fork",
            "--message=Created by `brew bump`", "--version=#{new_version}", name
   end
 end
